### PR TITLE
Remove "one line summary" from pr template

### PR DESCRIPTION
**Pull request by kolibril13**
We changed the method to create the changelog from using the `one line summary`  to the method described here https://github.com/ManimCommunity/manim/pull/986#issue-564730313 (automatically taking the title from the pr).
So there is no point in still having the `one line summary` in the pr template.
Also, I think we can delete this wiki page: https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release

### CODE REVIEW

**Review by jsonvillanueva — APPROVED** [review]
Although I think we should discuss https://github.com/twisted/towncrier as well as making more informative squash and merge commit messages to alleviate pressure in generating changelogs.

**Comment by behackl** [comment]
I do find the oneline summaries very useful: yesterday, when editing the changelog, I replaced unhelpful merge commit messages by the corresponding summaries.

What I would rather propose, instead of removing the field altogether: Developers merging PRs should take the summary provided by the PR author, and use that as the merge message.

I do agree that the reference to the wiki can be removed.

**Comment by naveen521kk** [comment]
For me I think it should be better that changelog should be written by people who author them. Only then things would be looking good. 

> Although I think we should discuss https://github.com/twisted/towncrier

This seems to be a nice Idea, I suggested in discord...
What this does is contributors will write changelog in a new file with their PR number and what type of PR in the filename rST format, and that tool will merge them into a single file when releasing. That would be like how we were doing previously but would avoid merge conflicts. I have this is also used in many other projects.

**Comment by jsonvillanueva** [comment]
> > Although I think we should discuss https://github.com/twisted/towncrier
> 
> This seems to be a nice Idea, I suggested in discord...

I wasn't trying to steal your thunder, merely redirect it. I know @behackl has expressed his thoughts on it in the 0.3.0 PR release and you mentioned it again in @eulertour's first 0.4.0 draft release, but I'm not sure if they've changed since.

I haven't read much into towncrier, but if it makes the process of creating a release *easier* for the mods who tend to make the release then I'm all for it. Ideally, the release procedure should be to simply bump the version number (and an end-user oriented changelog should be autogenerated). 

It looks like towncrier would be good for Manim, but it does involve being more stringent about some of our practices, in particular:
> `towncrier` works best in a development system where all merges involve closing a ticket.

**Comment by jsonvillanueva** [comment]
> I do find the oneline summaries very useful: yesterday, when editing the changelog, I replaced unhelpful merge commit messages by the corresponding summaries.

I think this is a responsibility that shouldn't be placed on the person making the release, but instead the one who squash-merged the PR in the first place. As you've said:
> Developers merging PRs should take the summary provided by the PR author, and use that as the merge message.

Unfortunately, us Devs are fallible and I think it's *far* easier to ensure each PR has an issue associated with it since more all Devs (and members in general) can see if an issue is linked than to entrust a single Dev to copy/paste the corresponding summary.

However we wish to approach this, the process should be made easier for the releaser. There shouldn't need to be a 2-3 day delay on the release due to changelog matters. Either way, us Devs need to be a bit more considerate in our git practices, either in: creating issues for PRs that aren't associated with one for towncrier, or transferring the better summary into the merge message.

**Comment by eulertour** [comment]
I don't see much value in maintaining a changelog beyond just taking the list of PR titles and ensuring said titles are descriptive. There are a few reasons for this:
1. The average Git repo either doesn't have one or doesn't do anything more than that, so people don't rely on changelogs
2. Manually compiling the changelog at release time won't scale, so the solution will gradually work worse over time
3. The alternative of searching through commits, using `git blame`, and searching through code is more accurate and always reliable

> There shouldn't need to be a 2-3 day delay on the release due to changelog matters.

I'd go a step further and say that if there's such a delay then we need to fix the process. Not only does it stop us from getting our new code to PyPI but it also complicates the process of managing the PRs that cluster around release time.

So basically anything we choose should have the effort distributed at commit time and only a constant (i.e. automatable) amount of work at release time.

**Comment by jsonvillanueva** [comment]
>     1. The average Git repo either doesn't have one or doesn't do anything more than that, so people don't rely on changelogs
>     2. Manually compiling the changelog at release time won't scale, so the solution will gradually work worse over time
>     3. The alternative of searching through commits, using `git blame`, and searching through code is more accurate and always reliable

1. I expect ManimCommunity/manim to be better than the average repo. We have all these great minds working together after all 😄
2. Agreed.
3. While true, I don't expect end-users to know how to use git. The changelog is for a general audiences.

> So basically anything we choose should have the effort distributed at commit time and only a constant (i.e. automatable) amount of work at release time.

Agreed. That said, I think *some* sort of automated changelog generator is in order; it should allow *multiple* Devs the opportunity to create descriptive changelogs (e.g. based on the title of the PR, the issue it closes, etc). If not towncrier, then what?

**Comment by eulertour** [comment]
Is this ready to merge now?

**Comment by jsonvillanueva** [comment]
I think so? I'd wait for @behackl to approve/chime in.

Also, looking at this again, I think we should edit the PR_template [at the top](https://github.com/ManimCommunity/manim/pull/1073/files#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35R1-R6) to tell the author of the PR to make their title descriptive.

**Comment by behackl** [comment]
I'd like to keep the field where authors provide a short summary themselves as long as we don't switch to, e.g., towncrier.

**Comment by behackl** [comment]
... I'd also be okay with a corresponding addition that the PR title *should be* expressive enough so that it can serve as the summary, as suggested by @jsonvillanueva.

**Comment by jsonvillanueva** [comment]
I made this template. How does this sound?
```
<!--
Thank you for your interest in contributing to ManimCommunity!

Before filling out the details below, please ensure:
- Your local changes are up-to-date with ManimCommunity/manim's latest
changes.

- The title of your PR starts with a keyword:
  Feature:
  Bugfix:
  Enhancement:
  Doc:
  Removal:
  Misc:
  
  Or create a keyword if none of the previous keywords are applicable.
  I
- The title of your PR gives a short summary to end-users on what changes
have been made. Some examples:
  - `Bugfix: Fixed the issue with last animations not running to completion`)
  - `Feature: Added support for SVG gradients`)
-->
```

**Comment by jsonvillanueva** [comment]
> 
> 
> ... I'd also be okay with a corresponding addition that the PR title _should be_ expressive enough so that it can serve as the summary, as suggested by @jsonvillanueva.

@behackl Would you be okay in removing the oneline summary if the above intro were written in? I'd like to commit this in now so that other's can modify/edit it if it's favorable.

**Comment by behackl** [comment]
> > ... I'd also be okay with a corresponding addition that the PR title _should be_ expressive enough so that it can serve as the summary, as suggested by @jsonvillanueva.
> 
> @behackl Would you be okay in removing the oneline summary if the above intro were written in? Sorry if that's what you meant by this, I just read it as keep the summary and also add the part about making more descriptive titles. I'd like to commit this in now so that others can modify/edit it if it's favorable.

Yes, that would be okay for me. If the PR title overtakes its role, the oneline summary field can of course be removed.

**Comment by jsonvillanueva** [comment]
Feel free to merge/fix this PR's title.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (kolibril13) [inline .github/PULL_REQUEST_TEMPLATE.md:6]:
```suggestion
Please ensure that your pull request works with the latest
version of ManimCommunity/manim.
```
@jsonvillanueva: I d like to make this first sentence as short as possible, so the chance that people will read the following keyword section is higher.

Comment (jsonvillanueva) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: I'd prefer to keep the bullet-point nature of this section but agree it could be more concise. Maybe this should also be part of the checklist in the [# Acknowledgement](https://github.com/ManimCommunity/manim/pull/1073/files#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35R44) heading?

Comment (jsonvillanueva) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: After looking at how[ NumPy does it](https://numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message), we could also have this in our contributing guidelines on the docs since the `# Acknowledgement` section links to it.

Comment (jsonvillanueva) [reply .github/PULL_REQUEST_TEMPLATE.md]: I've made it shorter, adopted NumPy acronyms, and added the checklist items. Lmk what you think @kolibril13

Comment (kolibril13) [reply .github/PULL_REQUEST_TEMPLATE.md]: > I'd prefer to keep the bullet-point nature

I agree, lets do it like you said.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (kolibril13) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Personally, I don't like these acronyms too much, as it is another learning step as a contributor, and it would take some time to enforce the usage of them.
Additionally,  I think they are a bit redundant with the pr labels.
But in case we don't have a good alternative, I am fine with using them

Comment (jsonvillanueva) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: I was just modeling after NumPy... after some quick research into *why* they do this in their repo, it's because they have an [automatic labeler workflow](https://github.com/numpy/numpy/blob/main/.github/workflows/labeler.yml). I agree this can be removed. The [changelog generator script](https://github.com/numpy/numpy/blob/main/tools/changelog.py) NumPy uses doesn't actually take advantage of these acronyms either which seems like a lost opportunity.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (behackl) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: The content of this string is actually not markdown, no need for code literals there.

```suggestion
  - Fixed last animations not running to completion
  - Added gradient support and documentation for SVG files
  Examples of what *NOT* to do:
  - "fixed that styling issue" - not descriptive enough
  - "fixed issue #XYZ" - end-user needs to do further research
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (behackl) [inline .github/PULL_REQUEST_TEMPLATE.md:37]: The decision whether the PR title is descriptive enough or not is on the reviewer. It's still a good idea to have a checkbox for PR authors as well to make them aware of the issue. Suggestion for slight rewording:

```suggestion
- [ ] I have chosen a descriptive PR title (see top of PR template for examples)
```

**Review by behackl — APPROVED** [review]
This looks good to me now; I've just added two minor suggestions.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (kolibril13) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: Sounds great!

**Comment by kolibril13** [comment]
Nice, in case that there are no further comments, I will merge this in a few hours, and delete this wiki page:
https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release

**Comment by behackl** [comment]
> Nice, in case that there are no further comments, I will merge this in a few hours, and delete this wiki page:
> https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release

In that case please also remove the reference to the wiki page in `changelog.rst`.

**Comment by kolibril13** [comment]
> In that case please also remove the reference to the wiki page in `changelog.rst`.

good point!
This is done now.

**Pull request merged**
